onnxruntime/core/framework/func_kernel.h

```diff
@@ -57,15 +68,12 @@ class FunctionKernel : public OpKernel {
       return Status(common::ONNXRUNTIME, common::FAIL, "FuncKernel call failed with error code: " + std::to_string(ret));
 
     for (int i = 0; i < num_outputs_; i++) {
+      //since we don't want to re-allocate te output buffer, create a tensor with empty buffer in execution frame first.
+      Tensor* output = context->Output(i, {0});
       TensorShape output_shape(std::vector<int64_t>(output_tensors[i].shape, output_tensors[i].shape + output_tensors[i].ndim));
-      Tensor* output = context->Output(i, output_shape);
-      auto data = output->MutableDataRaw();
-      //TODO: for string tensors, this copy is not correct.
-      ORT_ENFORCE(output->DataType() != DataTypeImpl::GetType<std::string>());
-      memcpy(data, output_tensors[i].data, output->DataType()->Size() * output_shape.Size());
-      //Release output tensors (buffer, shape).
-      host_allocator_->Free(output_tensors[i].data);
-      // for shape, becauset the TempSpaceAllocator we use could be a device allocator, if the kernel is assigned to a device like gpu.
+      //swap the buffer created by compute method into the output tensor.
+      ORT_RETURN_IF_ERROR(output->ReplaceBuffer(&output_tensors[i], output_allocator_));
```

Comment:
>ORT_RETURN_IF_ERROR(output->ReplaceBuffer(&output_tensors[i], output_allocator_)); [](start = 6, length = 82)

Would this work with allocation planner?

Reply:
I assume if we fused the nodes and do the compile, memory planner won't trying to reuse other buffer for this node's output, because we didn't provide the output shape in the static plan, so allocation planner can't do memory sharing. I can add more check for it.